packages/js/product-editor/src/contexts/validation-context/README.md

```diff
@@ -0,0 +1,39 @@
+# Validations and error handling
+
+This directory contains some components and hooks used for validations in the product editor.
+
+See [Product Editor Development Handbook](../../../../../../docs/product-editor-development/README.md)
+for more information.
+
+## Add a link to error snackbar
+
+### What happens when there is an error in a field?
+
+1. Every field [will be validated](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/product-editor/src/contexts/validation-context/validation-provider.tsx#L87-L110).
+
+2. An object with the [fields with errors](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/product-editor/src/contexts/validation-context/validation-provider.tsx#L74) will be returned.
+
+3. If, for instance, the name field is empty, the validation will fail and will throw an object like this:
+
+    ```javascript
+    { message: 'Product name is required.'; context: [block id]; validatorId: [prefix + block id] }
+    ```
+
+4. This is the result of the [name validator](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/product-editor/src/blocks/product-fields/name/edit.tsx#L84-L101) and the [validatorId addition](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/product-editor/src/contexts/validation-context/validation-provider.tsx#L69).
```

Comment:
This should also be a sub-bullet of step 2, but instead of linking the code we may want to list it as an example:
```
async function nameValidator() {
			if ( ! name || name === AUTO_DRAFT_NAME ) {
				return {
					message: __( 'Product name is required.', 'woocommerce' ),
					context: clientId,
				};
			}

			if ( name.length > 120 ) {
				return {
					message: __(
						'Please enter a product name shorter than 120 characters.',
						'woocommerce'
					),
					context: clientId,
				};
			}
		},
```

Reply:
Done in [be38da9](https://github.com/woocommerce/woocommerce/pull/49723/commits/be38da928a6429fd51eed853c2fe3eebe15e30f9)